forte/data/data_store.py

```diff
@@ -1044,20 +1072,166 @@ def _create_new_entry(
         tid: int = self._new_tid() if tid is None else tid
         entry: List[Any] = []
 
-        entry.extend(attribute_data)
-
         entry += [tid, type_name]
         entry += self._default_attributes_for_type(type_name)
 
         return entry
 
+    def get_datastore_attr_idx(self, type_name: str, attr: str) -> int:
+        r"""This function returns the index of where a given attribute
+        `attr` is stored in the Data Store entry of type `type_name`
+
+        Args:
+            type_name (str): The fully qualified type name of the entry.
+            attr (str): The name of the attribute whose index needs to be fetched.
+
+        Returns:
+            An integer representing the attributes position in the Data
+            Store entry.
+        """
+        try:
+            return DataStore._type_attributes[type_name][
+                constants.ATTR_INFO_KEY
+            ][attr][constants.ATTR_INDEX_KEY]
+        except KeyError as e:
+            raise KeyError(
+                f"Attribute {attr} is not a part "
+                f"of type attributes for {type_name}"
+                f"or an entry of type {type_name} does not exist"
```

Comment:
Would an index out of bound happen here?

Reply:
I don't think so because we are accessing the position index from the `_type_attributes` dict which has all these keys set at all times.